src/wp-admin/includes/image.php

```diff
@@ -301,18 +310,42 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 			if ( ! is_wp_error( $resized ) ) {
 				// Append "-scaled" to the image file name. It will look like "my_image-scaled.jpg".
 				// This doesn't affect the sub-sizes names as they are generated from the original image (for best quality).
-				$saved = $editor->save( $editor->generate_filename( 'scaled' ) );
+				$saved = $editor->save( $editor->generate_filename( 'scaled' ), $primary_mime_type );
 
 				if ( ! is_wp_error( $saved ) ) {
 					$image_meta = _wp_image_meta_replace_original( $saved, $file, $image_meta, $attachment_id );
 
+					$image_meta['sources'][ $primary_mime_type ] = _wp_get_sources_from_meta( $saved );
+
 					// If the image was rotated update the stored EXIF data.
 					if ( true === $rotated && ! empty( $image_meta['image_meta']['orientation'] ) ) {
 						$image_meta['image_meta']['orientation'] = 1;
 					}
+					wp_update_attachment_metadata( $attachment_id, $image_meta );
 				} else {
 					// TODO: Log errors.
 				}
+
+				// Save additional mime types.
+				foreach ( $additional_mime_types as $output_mime_type ) {
+					$editor = wp_get_image_editor( $file );
+
+					if ( is_wp_error( $editor ) ) {
+						// This image cannot be edited.
+						continue;
+					}
+
+					if ( empty( $mime_extension_map[ $output_mime_type ] ) ) {
+						// Skip mime types that are not supported.
+						continue;
+					}
+
+					$saved = $editor->save( $editor->generate_filename( 'scaled', null, $mime_extension_map[ $output_mime_type ] ), $output_mime_type );
```

Comment:
This here doesn't work for all MIME types, `$mime_extension_map[ $output_mime_type ]` may have multiple extensions separated by a pipe (for example for `image/jpeg` it is `jpg|jpeg|jpe`. So we also need to make sure to extract the first extension in case there are multiple.
```suggestion
					$extension = explode( '|', $mime_extension_map[ $output_mime_type ] )[0];
					$saved     = $editor->save( $editor->generate_filename( 'scaled', null, $extension ), $output_mime_type );
```

It may be even better to do extract this data in a single central place above though, so that this logic doesn't run in every `foreach` cycle. You could already do this at the top where you're also calling `_wp_get_primary_and_additional_mime_types()`, based on those MIME types.

Reply:
👍🏼